haskell/protobuf.bzl

```diff
@@ -201,12 +201,14 @@ _haskell_proto_aspect = aspect(
 )
 
 def _haskell_proto_library_impl(ctx):
-    dep = ctx.attr.deps[0]  # FIXME
-    return [
-        dep[HaskellBuildInfo],
-        dep[HaskellLibraryInfo],
-        DefaultInfo(files = dep[HaskellProtobufInfo].files),
-    ]
+    res = []
+    defaultInfoFilesDepset = depset()
+    for dep in ctx.attr.deps:
+        res.append(dep[HaskellBuildInfo])
```

Comment:
I don't think this will work like you intend.  As far as I know, you can only return a single provider from a rule.  For example, you'd need to return a single HaskellLibraryInfo that merged all of the dependencies' HaskellLibraryInfos into one.